### PR TITLE
[stable/kube-state-metrics] Add option for service annotations

### PR DESCRIPTION
**Pull request by coreypobrien**
#### What this PR does / why we need it:
Add option for custom service annotations. Useful for triggering other collection systems besides prometheus.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/kube-state-metrics/templates/service.yaml [hunk 1]
```diff
@@ -10,10 +10,13 @@ metadata:
...
 {{ toYaml .Values.customLabels | indent 4 }}
 {{- end }}
-  {{- if .Values.prometheusScrape }}
   annotations:
+    {{- if .Values.prometheusScrape }}
     prometheus.io/scrape: '{{ .Values.prometheusScrape }}'
-  {{- end }}
+    {{- end }}
+    {{- if .Values.service.annotations }}
+    {{- toYaml .Values.service.annotations | nindent 4 }}
```
Comment (tariq1890) [inline stable/kube-state-metrics/templates/service.yaml:18]: why `nindent`? Shouldn't `indent` do?

Comment (coreypobrien) [reply stable/kube-state-metrics/templates/service.yaml:18]: Just for cleaner reading (to keep the `{{- toYaml...` indented under the annotations)

Comment (tariq1890) [reply stable/kube-state-metrics/templates/service.yaml:18]: We also need to account for scenarios where both custom annotations and prom scrapes are disabled.

Comment (coreypobrien) [reply stable/kube-state-metrics/templates/service.yaml:18]: This should work correctly as is in that situation:

```
$ helm install --namespace asdf /Users/corey/src/charts/stable/kube-state-metrics --name asdf --set prometheusScrape=false
NAME:   asdf
LAST DEPLOYED: Thu Jul 25 00:18:36 2019
NAMESPACE: asdf
STATUS: DEPLOYED

RESOURCES:
==> v1/Deployment
NAME                     READY  UP-TO-DATE  AVAILABLE  AGE
asdf-kube-state-metrics  0/1    1           0          1s

==> v1/Pod(related)
NAME                                      READY  STATUS             RESTARTS  AGE
asdf-kube-state-metrics-5b45bcffd6-dqqjg  0/1    ContainerCreating  0         1s

==> v1/Service
NAME                     TYPE       CLUSTER-IP     EXTERNAL-IP  PORT(S)   AGE
asdf-kube-state-metrics  ClusterIP  100.70.179.62  <none>       8080/TCP  1s

==> v1/ServiceAccount
NAME                     SECRETS  AGE
asdf-kube-state-metrics  1        1s

==> v1beta1/ClusterRole
NAME                     AGE
asdf-kube-state-metrics  1s

==> v1beta1/ClusterRoleBinding
NAME                     AGE
asdf-kube-state-metrics  1s


NOTES:
kube-state-metrics is a simple service that listens to the Kubernetes API server and generates metrics about the state of the objects.
The exposed metrics can be found here:
https://github.com/kubernetes/kube-state-metrics/blob/master/docs/README.md#exposed-metrics

The metrics are exported on the HTTP endpoint /metrics on the listening port.
In your case, asdf-kube-state-metrics.asdf.svc.cluster.local:8080/metrics

They are served either as plaintext or protobuf depending on the Accept header.
They are designed to be consumed either by Prometheus itself or by a scraper that is compatible with scraping a Prometheus client endpoint.


$ kubectl -n asdf get svc -o yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: Service
  metadata:
    creationTimestamp: 2019-07-25T04:18:36Z
    labels:
      app.kubernetes.io/instance: asdf
      app.kubernetes.io/managed-by: Tiller
      app.kubernetes.io/name: kube-state-metrics
      helm.sh/chart: kube-state-metrics-2.2.1
    name: asdf-kube-state-metrics
    namespace: asdf
    resourceVersion: "35535510"
    selfLink: /api/v1/namespaces/asdf/services/asdf-kube-state-metrics
    uid: 45b4092b-ae93-11e9-aad5-063a23d7243c
  spec:
    clusterIP: 100.70.179.62
    ports:
    - name: http
      port: 8080
      protocol: TCP
      targetPort: 8080
    selector:
      app.kubernetes.io/instance: asdf
      app.kubernetes.io/name: kube-state-metrics
    sessionAffinity: None
    type: ClusterIP
  status:
    loadBalancer: {}
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**Review by tariq1890 — APPROVED** [review]
/lgtm

**Comment by tariq1890** [comment]
Thanks @coreypobrien!

**Pull request merged**